### PR TITLE
refactor: use db pool in settings repository

### PR DESCRIPTION
**Pull request by jan-zabloudil**

### CODE REVIEW

File: service/settings.go [hunk 1]
```diff
@@ -34,16 +34,14 @@ func (s *SettingsService) Update(ctx context.Context, u model.UpdateSettingsInpu
...
 	}
 
-	settings, err := s.Get(ctx, authUserID)
-	if err != nil {
-		return model.Settings{}, err
-	}
+	settings, err := s.repository.Update(ctx, func(s model.Settings) (model.Settings, error) {
+		if err := s.Update(u); err != nil {
+			return model.Settings{}, apierrors.NewSettingsUnprocessableError().Wrap(err).WithMessage(err.Error())
+		}
```
Comment (jan-zabloudil) [inline service/settings.go:40]: Reading settings and updating it is done in one transaction.

Comment (bolshakoff) [reply service/settings.go:40]: Repository returning new `settings` is fine. The rest also looks OK now.

Comment (bolshakoff) [reply service/settings.go:40]: (optional) Regarding one transaction, good ambition, and the current implementation is probably fine, albeit perhaps too low-level.
But do we really need it? What's the worst case and what are the potential consequences?
Or as a compromise, consider using `SELECT ... FOR UPDATE`.

**Review by bolshakoff — APPROVED** [review]

File: service/settings.go (hunk 1 above)
Comment (jan-zabloudil) [reply service/settings.go:40]: Thanks 👍 I've decided to try the approach with transactions to practice it. I'm going to keep the current implementation, but I'll add a comment to consider a implementation you suggested. Once I gain more experience while implementing the repository for this app, I might switch to a simpler approach.

**Pull request merged**